weather_getter/app.py

```diff
@@ -2,22 +2,56 @@
 from datetime import datetime
 from geopy.geocoders import Nominatim
 import json
+import sys
+import re
 
 
 city_name = input("Please Enter City Name:")
 
+if re.fullmatch(r'^[a-zA-Z\s]+$',city_name) == None:
+    print("The city name is invalid.")
+    sys.exit()
+
 geo_locator = Nominatim(user_agent="EB_weather_getter")
 location = geo_locator.geocode(city_name)
-latitude = location.latitude
-longitude = location.longitude
 
-url = f'https://api.open-meteo.com/v1/forecast?latitude={latitude}&longitude={longitude}&hourly=temperature_2m'
+if location:
+    if hasattr(location,'latitude'):
+        latitude = location.latitude
+    else:
+        latitude = ""
+    
+    if hasattr(location,'longitude'):
+        longitude = location.longitude
+    else:
+        latitude = ""
```

Comment:
What happened if latitude set to ""? Does open-meteo require floating point latitude or latitude?

Reply:
what do you mean?